src/linear/coordinate_common.h

```diff
@@ -64,6 +65,9 @@ inline double CoordinateDelta(double sum_grad, double sum_hess, double w,
  * \return  The weight update.
  */
 inline double CoordinateDeltaBias(double sum_grad, double sum_hess) {
+  if (std::fpclassify(sum_hess) == FP_ZERO) {
+    return 0;
+  }
```

Comment:
The division may still be invalid even when `sum_hess` is a small nonzero value, due to overflow. We should just compute `-sum_grad / sum_hess` and then check if the resulting value is infinity, using `std::isinf`.